invoices/settlement_interceptor.go

```diff
@@ -0,0 +1,267 @@
+package invoices
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/lntypes"
+)
+
+// SafeCallback is a thread safe wrapper around an InterceptorClientCallback.
```

Comment:
Where's `InterceptorClientCallback` defined?

Reply:
Note that for this we also need to be able to feed in the wire HTLC TLV blobs. Today we don't store those on disk for accepted invoice HTLCs: https://github.com/lightningnetwork/lnd/blob/aee029483a9c1aa9c6851bfdfbf084e1919363a9/channeldb/invoices.go#L1981-L2002

Reply:
> Where's `InterceptorClientCallback` defined?

In the same package.

Reply:
The invoice settlement interceptor and interceptor interface are now in the same commit.